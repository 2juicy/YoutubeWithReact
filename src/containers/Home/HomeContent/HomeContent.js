import React from "react";
import "./HomeContent.scss";
import VidGrid from "../../../components/VidGrid/VidGrid";
import {
  getMostPopularVideos,
  getVideosByCategory
} from "../../../store/reducers/video";
import { connect } from "react-redux";

function HomeContent(props) {
  const AMOUNT_TRENDING_VIDEOS = 12;
  const trendingVideos = props.mostPopularVideos.slice(
    0,
    AMOUNT_TRENDING_VIDEOS
  );

  return (
    <div className="home-content">
      <div className="responsive-vid-grid-container">
        <VidGrid title="Trending" videos={trendingVideos} />
      </div>
    </div>
  );
}

function mapStateToProps(state) {
  return {
    // videosByCategory: getVideosByCategory(state),
    mostPopularVideos: getMostPopularVideos(state)
  };
}

export default connect(mapStateToProps, null)(HomeContent);
