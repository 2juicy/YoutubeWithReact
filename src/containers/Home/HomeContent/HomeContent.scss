@import "../../../styles/shared.scss";

.home-content {
  margin: $header-height 0 0 $sidebar-width;
  font-family: Roboto, Arial, sans-serif;
}

.responsive-vid-grid-container {
  max-width: 95%;
  margin: 0 auto;
}
