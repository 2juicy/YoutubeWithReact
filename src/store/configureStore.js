import { applyMiddleware, createStore } from "redux";
import reducer from "./reducers";
import createSagaMiddleware from "redux-saga";
import rootSaga from "./sagas";

export function configureStore() {
  const sagaMiddleware = createSagaMiddleware;
  const store = createStore(reducer, applyMiddleware(sagaMiddleware));
}
