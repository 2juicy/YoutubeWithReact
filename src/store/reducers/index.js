import apiReducer from "./api";
import { combineReducers } from "redux";
import videosReducer from "./video";

export default combineReducers({
  api: apiReducer,
  videos: videosReducer
});
