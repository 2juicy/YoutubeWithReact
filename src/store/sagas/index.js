import { all, call, put, fork } from "redux-saga/effects";
import {
  watchMostPopularVideos,
  watchMostPopularVideosByCategory,
  watchVideoCategories
} from "./video";

export default function*() {
  yield all([
    fork(watchMostPopularVideos),
    fork(watchVideoCategories),
    fork(watchMostPopularVideosByCategory)
  ]);
}

export function* fetchEntity(request, entity, ...args) {
  try {
    const res = yield call(request);
    yield put(entity.success(res.result, ...args));
  } catch (err) {
    yield put(entity.failure(err, ...args));
  }
}

export function ignoreErrors(fn, ...args) {
  return () => {
    const ignoreErrorCallback = res => res;
    return fn(...args).then(ignoreErrorCallback, ignoreErrorCallback);
  };
}
