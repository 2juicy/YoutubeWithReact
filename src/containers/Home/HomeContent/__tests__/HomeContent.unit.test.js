import { shallow } from "enzyme";
import HomeContent from "../HomeContent";
import React from "react";
import { Provider } from "react-redux";
import { configureStore } from "../../../../store/configureStore";

const store = configureStore();

describe("HomeContent", () => {
  test("renders", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <HomeContent />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
