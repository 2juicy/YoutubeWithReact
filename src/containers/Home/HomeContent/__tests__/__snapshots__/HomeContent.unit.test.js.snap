// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContent renders 1`] = `
<div
  className="home-content"
>
  <div
    className="responsive-vid-grid-container"
  >
    <VidGrid
      title="Trending"
    />
    <VidGrid
      title="Music"
    />
    <VidGrid
      title="Anime"
    />
  </div>
</div>
`;
